### PR TITLE
Updated ICDS UCR Tests

### PR DESCRIPTION
**Pull request by sheelio**
This adds new tests for the child health monthly data source.  It also refactors the custom UCR expressions and the test structure to be a bit cleaner.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 9 errors:

* custom/icds_reports/ucr/expressions.py, line 21 - E501 line too long (116 > 115 characters)
* custom/icds_reports/ucr/tests/base_test.py, line 73 - E226 missing whitespace around arithmetic operator
* custom/icds_reports/ucr/tests/base_test.py, line 75 - E226 missing whitespace around arithmetic operator
W292 no newline at end of file
* custom/icds_reports/ucr/tests/test_ccs_record_monthly.py, line 1078 - W292 no newline at end of file
* custom/icds_reports/ucr/tests/test_child_health_monthly.py, line 918 - E124 closing bracket does not match visual indentation
* custom/icds_reports/ucr/tests/test_child_health_monthly.py, line 980 - E231 missing whitespace after ','
* custom/icds_reports/ucr/tests/test_child_health_monthly.py, line 1004 - E124 closing bracket does not match visual indentation
* custom/icds_reports/ucr/tests/test_child_health_monthly.py, line 1094 - E124 closing bracket does not match visual indentation
* custom/icds_reports/ucr/tests/test_child_health_monthly.py, line 1126 - W292 no newline at end of file

**Comment by sheelio** [comment]
@esoergel 

I think you reviewed the last set of tests.  Can you give this set a look?

File: custom/icds_reports/ucr/tests/base_test.py [hunk 1]
```diff
@@ -60,3 +62,13 @@ def _rebuild_table_get_query_object(self):
...
+
+    def _run_iterative_monthly_test(self, case_id, cases, start_date=date(2015, 12, 1)):
+        query = self._rebuild_table_get_query_object().filter_by(doc_id=case_id)
+        self.assertEqual(query.count(), 7)
+
+        for index, test_values in cases:
+            row = query.all()[index]._asdict()
+            self.assertEqual(row['month'], start_date+relativedelta(months=index))
+            for key, value in test_values:
+                self.assertEqual(row[key], value)
```
Comment (esoergel) [inline custom/icds_reports/ucr/tests/base_test.py]: The python builtin `enumerate` would make these test cases a bit easier to write (it'd add the indices)
```python
In [1]: for index, val in enumerate(['a', 'b', 'c']):
   ...:     print index, val
   ...:     
0 a
1 b
2 c
```
Or better yet, just drop the `index` business altogether with `zip`:
```python
In [2]: tests = ['test1', 'test2', 'test3']

In [3]: real = ['real1', 'real2', 'real3', 'real4']

In [4]: for test, real in zip(tests, real):
   ...:     print test, real
   ...:     
test1 real1
test2 real2
test3 real3
```

File: custom/icds_reports/ucr/tests/test_child_health_monthly.py [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+        case.attrib['date_modified'] = form_date.isoformat()
+        case.attrib['case_id'] = case_id
+        case.attrib['xmlns'] = 'http://commcarehq.org/case/transaction/v2'
+        if nutrition_status is not None:
+            case_update = ElementTree.Element('update')
+            case_update.append(create_element_with_value('zscore_grading_wfa', nutrition_status))
+            case.append(case_update)
+        form.append(case)
+
+        submit_form_locally(ElementTree.tostring(form), self.domain, **{})
```
Comment (esoergel) [inline custom/icds_reports/ucr/tests/test_child_health_monthly.py]: we really need to make a better way to generate a form...

**Review by esoergel — APPROVED** [review]
I'm not gonna pretend that I read all 13,000 lines changed, but I skimmed it, feel like I got the gist.  It looks it's mostly cleanup and abstraction, which I'm down with.

**Pull request merged**